src/Storages/ExternalStream/Kafka/KafkaSource.cpp

```diff
@@ -49,6 +49,8 @@ KafkaSource::KafkaSource(
     , virtual_col_types(header.columns(), nullptr)
     , ckpt_data(consume_ctx)
 {
+    is_streaming = true;
```

Comment:
Look like something is missing, this flag shall be setup by framework atomically.  @yl-lisen

Reply:
We need to specify whether the current source is streaming, then other downstream pipes will automatically be set according to their input

Reply:
1. hmm, i think we may need comment this protocol in the code in the base class, like 
```
/// If derived children is a streaming source, derived children need explicitly set is_streaming = true
/// to enable streaming processing.
```
2. I don't get it why we will need explicitly set this up since the storage shall indicate if it is streaming or not and the corresponding source will inherit this property from the underlying storage atomically.

Reply:
Agree it, the better way is that set it in `Storage::read(...)`, some sources such as `stream` or `external stream` need set the streaming flag of their built Source.

Reply:
Or define an IStreamingSource(), which only set `is_streaming = true` in ctor. and some sources enabled streaming processing can inherit it

Reply:
Let's rethink about it in another PR. merge this one first.